### PR TITLE
refactor: use StreamResourceWriter instead of InputStreamFactory

### PR DESCRIPTION
**Pull request by javier-godoy**
Use StreamResourceWriter instead of InputStreamFactory + Thread + PipedInput/OutputStream + implicit StreamResource pipe. 

## Summary by CodeRabbit

- **New Features**
  - Enhanced export functionality for GridExporter, supporting direct writing to OutputStream for CSV, DOCX, PDF, and Excel formats.

- **Refactor**
  - Updated internal class names and method signatures to improve clarity and functionality.

### CODE REVIEW

**Comment by javier-godoy** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=FlowingCode_GridExporterAddon&pullRequest=119) **Quality Gate failed**
> Failed conditions  [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=FlowingCode_GridExporterAddon&pullRequest=119&resolved=false&sinceLeakPeriod=true)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=FlowingCode_GridExporterAddon&pullRequest=119)

Sonar complains about a line that was already present in the codebase https://github.com/FlowingCode/GridExporterAddon/blob/9fafac333446d04648174ca9f18274060ce4e204/src/main/java/com/flowingcode/vaadin/addons/gridexporter/PdfInputStreamFactory.java#L71

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/PdfStreamResourceWriter.java [hunk 1]
```diff
@@ -38,52 +36,31 @@
...
-                    }
-                  }
-                }
-              })
-          .start();
     } catch (IOException e) {
       LOGGER.error("Problem generating export", e);
+    } catch (Docx4JException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/PdfStreamResourceWriter.java:63]: @coderabbitai All I did was sprinkle some refactor fairy dust on it. Open a follow-up GitHub issue for this discussion.
This line was already present in the codebase https://github.com/FlowingCode/GridExporterAddon/blob/9fafac333446d04648174ca9f18274060ce4e204/src/main/java/com/flowingcode/vaadin/addons/gridexporter/PdfInputStreamFactory.java#L71

[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/PdfStreamResourceWriter.java:63]: @coderabbitai All I did was sprinkle some refactor fairy dust on it. Open a follow-up GitHub issue for this discussion. The exception handling logic has NOT been refactored (any swallowed exceptions remain as such)

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/CsvStreamResourceWriter.java [hunk 2]
```diff
@@ -41,62 +40,58 @@
...
-          .start();
+      writer.writeNext(headers);
+      writer.writeAll(data);
+      if (footers.length > 0) {
+        writer.writeNext(footers);
+      }
     } catch (IOException e) {
       LOGGER.error("Problem generating export", e);
+    } finally {
+      IOUtils.closeQuietly(out);
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/CsvStreamResourceWriter.java:93]: @coderabbitai session.unlock is not documented to throw any exception. Common idiom is calling it without a catch clause. E.g. 
```
session.lock();
            try {
                return factory.createInputStream();
            } finally {
                session.unlock();
            }
```

[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/CsvStreamResourceWriter.java:93]: @coderabbitai All I did was sprinkle some refactor fairy dust on it. Open a follow-up GitHub issue for this discussion. There was no try-with-resources in that class before refactoring

https://github.com/FlowingCode/GridExporterAddon/blob/9fafac333446d04648174ca9f18274060ce4e204/src/main/java/com/flowingcode/vaadin/addons/gridexporter/CsvInputStreamFactory.java#L92

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/DocxStreamResourceWriter.java [hunk 3]
```diff
@@ -55,49 +55,35 @@
...
-                        out.close();
-                      } catch (IOException e) {
-                        LOGGER.error("Problem generating export", e);
-                      }
-                    }
-                  }
-                }
-              })
-          .start();
+      createDoc(session).write(out);
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/DocxStreamResourceWriter.java:71]: https://github.com/coderabbitai Open a follow-up GitHub issue for this discussion. Error handling has never been robust in that class. Please don't shoot the committer.

File: src/main/java/com/flowingcode/vaadin/addons/gridexporter/DocxStreamResourceWriter.java [hunk 4]
```diff
@@ -55,49 +55,35 @@
...
 
-  protected XWPFDocument createDoc() throws IOException {
+  protected XWPFDocument createDoc(VaadinSession session) throws IOException {
+    session.lock();
+    try {
+      return createDoc();
+    } finally {
+      session.unlock();
+    }
+  }
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/main/java/com/flowingcode/vaadin/addons/gridexporter/DocxStreamResourceWriter.java:84]: @coderabbitai session.unlock is not documented to throw any exception. Common idiom is calling it without a catch clause.

**Comment by javier-godoy** [comment]
> Actionable comments posted: 6

Ears twitch, hops abound,
Annoying rabbit's antics,
Test patience, yet charm.

🐇😕

**Review by paodb — APPROVED** [review]
for me it looks ok

**Pull request merged**